package.json

```diff
@@ -38,31 +38,32 @@
     "size-limit": "size-limit"
   },
   "peerDependencies": {
-    "@stoplight/markdown-viewer": "^5.0.0",
-    "@stoplight/mosaic": "^1.0.0",
-    "@stoplight/mosaic-code-viewer": "^1.0.0",
+    "@stoplight/markdown-viewer": "^5",
+    "@stoplight/mosaic": "^1",
+    "@stoplight/mosaic-code-viewer": "^1",
     "react": ">=16.8",
     "react-dom": ">=16.8"
   },
   "dependencies": {
     "@fortawesome/free-solid-svg-icons": "^5.15.2",
-    "@stoplight/json": "^3.10.0",
+    "@stoplight/json": "^3.17.1",
     "@stoplight/json-schema-tree": "^2.1.2",
     "@stoplight/react-error-boundary": "^2.0.0",
     "@types/json-schema": "^7.0.7",
     "classnames": "^2.2.6",
+    "jotai": "^1.4.5",
```

Comment:
meh, do we really need yet another library?
I know we probably use it somewhere in our ecosystem, but given it's a library meant to be consumed by others that may not have it installed it'd be better to rely on contexts.
Is there any reason we cannot use contexts here?

Reply:
Yeah we have jotai in the rest of our ecosystem, and it's 2.6kb gzipped 🤷‍♂️. I had thought about context, but it adds then need to start bringing in React.useReducer and it becomes a larger change to the options context object we already have (at least how I think about mutable react context, there is prob a way to keep things almost the same). I was also hoping to improve the expanded/collapsed state in a future PR so that it remembers which children are expanded/collapsed when a parent is toggled, and figured could use something like jotai to help with that as well. What do you think?